### PR TITLE
Further fixes for servlet buckets

### PR DESCRIPTION
**Pull request by isaacrivriv**
- Added logic for waiting on chain startup
- Removed amount of logging for clean up and test updates
- Updated tests to properly work with netty changes
- Added logic to make tests properly work

### CODE REVIEW

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/inbound/HttpInboundServiceContextImpl.java [hunk 1]
```diff
@@ -681,6 +681,7 @@ public void sendResponseHeaders() throws IOException, MessageSentException {
 
         if (getHttpConfig().useNetty()) {
             sendHeaders(this.nettyResponse);
+            this.nettyContext.channel().flush();
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/inbound/HttpInboundServiceContextImpl.java:684]: Check this later on

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/NettyChain.java [hunk 2]
```diff
@@ -365,37 +249,35 @@ public synchronized void startNettyChannel() {
...
                     parent.chainState.set(ChainState.STARTED.val);
-
                     parent.serverChannel = f.channel();
                     VirtualHostMap.notifyStarted(owner, () -> currentConfig.getResolvedHost(), currentConfig.getConfigPort(), isHttps);
                     String topic = owner.getEventTopic() + HttpServiceConstants.ENDPOINT_STARTED;
                     postEvent(topic, currentConfig, null);
                 }
             });
+            if (restarting)
+                channelFuture.get(nettyFramework.getDefaultChainQuiesceTimeout(), TimeUnit.MILLISECONDS).await();
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/NettyChain.java:274]: Verify if await is necessary

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/message/NettyRequestMessage.java [hunk 3]
```diff
@@ -416,7 +414,7 @@ public String getVirtualHost() {
 
     @Override
     public int getVirtualPort() {
-
+        // TODO Missing proper implementation
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/message/NettyRequestMessage.java:417]: Implement logic for virtual port

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/BufferEncoder.java [hunk 4]
```diff
@@ -50,9 +50,11 @@ public void encode(ChannelHandlerContext context, AbstractMap.SimpleEntry<Intege
         // Check if "Content-Length" is set for this channel
         boolean hasContentLength = context.channel().hasAttr(NettyHttpConstants.CONTENT_LENGTH);
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/BufferEncoder.java:51]: Verify to remove this class

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/RemoteIpHandler.java [hunk 5]
```diff
@@ -117,6 +118,18 @@ private void setErrorState() {
...
+    }
+
+    public void resetState() {
+        MSP.log("reset state");
+        this.forwardedBy = null;
+        this.forwardedFor = null;
+        this.forwardedHost = null;
+        this.forwardedPort = null;
+        this.forwardedProto = null;
+        // Shouldn't this be true?
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/RemoteIpHandler.java]: Update this to true

**Review by mrsaldana — APPROVED** [review]
LGTM

**Pull request merged**